splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsOperation.java

```diff
@@ -215,4 +231,18 @@ public double getSampleFraction() {
     public boolean getUseSample() {
         return useSample;
     }
+
+    private ExecRow buildTemplateRow(DataTypeDescriptor[] dataTypeDescriptors)  throws StandardException{
+        ExecRow row = new ValueRow(dataTypeDescriptors.length);
+        int outputCol = 0;
+        for (DataTypeDescriptor dtd : dataTypeDescriptors) {
+            row.setColumn(outputCol + 1, dtd.getNull());
+            if (row.getColumn(outputCol + 1 ) instanceof SQLDecimal) {
+                ((SQLDecimal) row.getColumn(outputCol + 1)).setPrecision(dtd.getPrecision());
+                ((SQLDecimal) row.getColumn(outputCol + 1)).setScale(dtd.getScale());
+            }
+            outputCol++;
```

Comment:
The null SQLDecimal should have precision and scale, and you don't need to set it again

Reply:
Good catch, thanks @jyuanca .